onnxruntime/test/contrib_ops/attention_op_test.cc

```diff
@@ -1536,5 +1600,98 @@ TEST(AttentionTest, AttentionPrunedModel) {
+      3.14984703f, 0.10842596f, 4.25f, 5.65f,
+      3.14984703f, 0.10842596f, 4.25f, 5.65f,
+      3.96724534f, 0.07324841f, 4.25f, 5.65f};
+
+  OpTester tester("Attention", 1, onnxruntime::kMSDomain);
+  tester.AddAttribute<int64_t>("num_heads", static_cast<int64_t>(number_of_heads));
+  tester.AddAttribute<int64_t>("unidirectional", static_cast<int64_t>(0));
+
+  std::vector<int64_t> input_dims = {batch_size, sequence_length, hidden_size};
+  std::vector<int64_t> weights_dims = {hidden_size, 3 * hidden_size};
+  std::vector<int64_t> bias_dims = {3 * hidden_size};
+
+  std::vector<int64_t> mask_index_dims = {2 * batch_size};
+  std::vector<int64_t> output_dims = {batch_size, sequence_length, hidden_size};
+
+  tester.AddInput<float>("input", input_dims, input_data);
+  tester.AddInput<float>("weight", weights_dims, weight_data, true);  // Trigger pre-packing
+  tester.AddInput<float>("bias", bias_dims, bias_data);
+  tester.AddOutput<float>("output", output_dims, output_data);
+  tester.AddInput<int32_t>("mask_index", mask_index_dims, mask_index_data);
+
+  auto allocator = test::AllocatorManager::Instance().GetAllocator(CPU);
+  auto p_tensor = std::make_unique<Tensor>(DataTypeImpl::GetType<float>(), TensorShape(weights_dims),
+                                           weight_data.data(), OrtMemoryInfo(CPU, OrtAllocatorType::OrtDeviceAllocator));
+  OrtValue weight;
+
+  weight.Init(p_tensor.release(), DataTypeImpl::GetType<Tensor>(),
+              DataTypeImpl::GetType<Tensor>()->GetDeleteFunc());
+
+  SessionOptions so;
+
+  // Set up weight as a shared initializer to be shared between sessions
+  ASSERT_EQ(so.AddInitializer("weight", &weight), Status::OK());
+
+  // We want all sessions running using this OpTester to be able to share pre-packed weights if applicable
+  tester.AddPrePackedSharedContainerToSessions();
+
+  size_t used_cached_pre_packed_weights_counter = 0;
+
+  // Pre-packing is limited just to the CPU EP for now and we will only test the CPU EP
+  // and we want to ensure that it is available in this build
+  auto cpu_ep = []() -> std::vector<std::unique_ptr<IExecutionProvider>> {
+    std::vector<std::unique_ptr<IExecutionProvider>> execution_providers;
+    execution_providers.push_back(DefaultCpuExecutionProvider());
+    return execution_providers;
+  };
+
+  // Session 1
+  {
+    auto ep_vec = cpu_ep();
+    tester.Run(so, OpTester::ExpectResult::kExpectSuccess, "", {},
+               nullptr, &ep_vec, {}, &used_cached_pre_packed_weights_counter);
+    ASSERT_EQ(used_cached_pre_packed_weights_counter, static_cast<size_t>(0));  // No pre-packed weights have been shared thus far
+  }
+
+  // Session 2
+  {
+    auto ep_vec = cpu_ep();
+    tester.Run(so, OpTester::ExpectResult::kExpectSuccess, "", {},
+               nullptr, &ep_vec, {}, &used_cached_pre_packed_weights_counter);
+    ASSERT_EQ(used_cached_pre_packed_weights_counter, static_cast<size_t>(1));  // One pre-packed weight has been shared thus far
+  }
+}
```

Comment:
x86, web assembly doesn't support prepack, so you check of prepack will fail

Reply:
Skipped checking for sharing if there was no pre-packing at all.